src/pages/workspaces/workspace/analysis/modals/AnalysisModal.test.ts

```diff
@@ -0,0 +1,399 @@
+    render(h(AnalysisModal, defaultGcpModalProps))
+
+    // Act
+    const button = screen.getByAltText(buttonAltText)
+
+    await userEvent.click(button)
+
+    // Assert
+    screen.getByText(expectedTitle)
+  })
+
+  it.each([
+    { fileType: 'Python 3' },
+    { fileType: 'R' },
+  ])('GCP - Creates a new $fileType for Jupyter when no apps or runtimes are present and navigates to environment creation.', async ({ fileType }) => {
+    // Arrange
+    const createMock = jest.fn()
+    const analysisMock: Partial<GoogleStorageContract['analysis']> = jest.fn(() => ({
+      create: createMock
+    }))
+    const googleStorageMock: Partial<GoogleStorageContract> = ({
+      analysis: analysisMock as GoogleStorageContract['analysis']
+    })
+
+    asMockedFn(GoogleStorage).mockImplementation(() => googleStorageMock as GoogleStorageContract)
+    render(h(AnalysisModal, defaultGcpModalProps))
+
+    // Act
+    await act(async () => {
+      const button = screen.getByAltText('Create new notebook')
+
+      await userEvent.click(button)
+
+      const fileTypeSelect = await screen.getByLabelText('Language *')
+      await userEvent.click(fileTypeSelect)
+
+      const selectOption = await screen.findAllByText(fileType)
+      await userEvent.click(selectOption[1])
+
+      const nameInput = screen.getByLabelText('Name of the notebook *')
+      await userEvent.type(nameInput, 'MyNewFile')
+
+      const createButton = await screen.findByText('Create Analysis')
+      await userEvent.click(createButton)
+    })
+
+    // Assert
+    screen.getByText('Jupyter Cloud Environment')
+    expect(createFunc).toHaveBeenCalled()
+  })
+
+  it('GCP - Creates a new file for Jupyter when a Jupyter runtime is present and does not navigate to cloud environment page.', async () => {
+    // Arrange
+    render(h(AnalysisModal, { ...defaultGcpModalProps, runtimes: [getGoogleRuntime()] }))
+
+    // Act
+    await act(async () => {
+      const button = screen.getByAltText('Create new notebook')
+
+      await userEvent.click(button)
+
+      const fileTypeSelect = await screen.getByLabelText('Language *')
+      await userEvent.click(fileTypeSelect)
+
+      const selectOption = await screen.findAllByText('Python 3')
+      await userEvent.click(selectOption[1])
+
+      const nameInput = screen.getByLabelText('Name of the notebook *')
+      await userEvent.type(nameInput, 'MyNewFile')
+
+      const createButton = await screen.findByText('Create Analysis')
+      await userEvent.click(createButton)
+    })
+
+    // Assert
+    expect(screen.queryByText('Jupyter Cloud Environment')).toBeNull()
+    expect(createFunc).toHaveBeenCalled()
+  })
+
+  it.each([
+    { fileType: 'R Markdown (.Rmd)' },
+    { fileType: 'R Script (.R)' }
+  ])('GCP - Creates a new $fileType for RStudio when no apps or runtimes are present and navigates to environment creation.', async ({ fileType }) => {
+    // Arrange
+    render(h(AnalysisModal, defaultGcpModalProps))
```

Comment:
nit: i think render is technically part of act, arrange is mainly for variables

Reply:
My thinking was in other tests we adjust some of the default values, so it's essentially setting up the initial render. In my mind, `defaultGcpModalProps` is essentially the "arrange"

Reply:
@msilva-broad has opinions on this.

It depends on the test. If the test is checking that something is rendered, render could be the "act". On the other hand, if the test is checking some user interaction after the initial render, render may be part of the "arrange".

Reply:
nick is correct.  //Arrange is not just about instantiating and setting up mocks... its to get the code into the "pre-condition" state before the main event.... so Nick's examples are spot on.